src/harness/security_testcase.py

```diff
@@ -140,6 +154,146 @@ def findIndexMatching(array, regex):
+          # Checking the validity of the date of the certificate is taken care by the pyopenssl v17.5.0 library
+          self.assertIsNotNone(cnotAfter)
+          logging.debug('Not After %s ', cnotAfter)
+          self.assertIsNotNone(cnotBefore)
+          logging.debug('Not Before %s ', cnotBefore)
+          logging.debug('Common Name ->subject %s ', crypto.X509Name(cert.get_subject()).commonName)
+          logging.debug('Common Name ->Issuer %s ', crypto.X509Name(cert.get_issuer()).commonName)
+          logging.debug('organizationName %s ', crypto.X509Name(cert.get_issuer()).organizationName )
+          logging.debug('ROLE  %s ', cert.get_extension(4))
+          logging.debug('CA  %s ', cert.get_extension(2))
+          self.assertEqual(cert.get_issuer().commonName, default_values["Issuer"])
+          self.assertEqual(cert.get_subject().commonName, default_values["Subject"])
+          self.assertEqual(cert.get_subject().countryName , default_values["countryName"])
+          self.assertEqual(cert.get_subject().stateOrProvinceName  , default_values["stateOrProvinceName"])
+          self.assertEqual(cert.get_subject().localityName   , default_values["localityName"])
+          self.assertEqual(cert.get_subject().organizationName   , default_values["organizationName"])
+          self.assertEqual(cert.get_subject().organizationalUnitName    , default_values["organizationalUnitName"])
+          self.assertEqual(cert.get_subject().commonName   , default_values["commonName"])
+          self.assertGreaterEqual(serial_number, 0,serial_number)
+          self.assertLess(len(str(serial_number)), 21)
+          self.assertEqual(cert.get_version(), 2)
+          self.assertGreaterEqual(pubkey.bits(), 2048)
+
+          policy_sas = str(cert.get_extension(5))
+          #The SAS UUT provided certificate does not contain the following custom SAS OIDs:
+          self.assertIn("1.3.6.1.4.1.46609.1.1",policy_sas)
+          self.assertNotIn("1.3.6.1.4.1.46609.1.2",policy_sas)
+          self.assertNotIn("1.3.6.1.4.1.46609.1.3",policy_sas)
+          self.assertNotIn("1.3.6.1.4.1.46609.1.4",policy_sas)
+          self.assertNotIn("1.3.6.1.4.1.46609.1.5",policy_sas)
+          self.assertNotIn("1.3.6.1.4.1.46609.1.7",policy_sas)
+          self.assertIn("1.3.6.1.4.1.46609.1.6",policy_sas, 0)
+          policy_ca = str(cert.get_extension(2))
+          self.assertIn("CA:FALSE",policy_ca)
+          self.assertNotIn("pathlen",policy_ca, 0)
+          logging.debug('ROLE SAS  %s ',policy_sas)
+          signature_alogo = cert.get_signature_algorithm()
+          signature_alog_list = ["sha256WithRSAEncryption","sha384WithRSAEncryption","sha512WithRSAEncryption","ecdsa-with-Sha256","ecdsa-with-Sha384","ecdsa-with-Sha512"]
+          self.assertIn(signature_alogo,signature_alog_list)
+          logging.info("All fields in certificates are validated successfully")
+        return ok 
```

Comment:
Basically I think that this function need more work to match the specification:
```
The SAS UUT response must satisfy all of the following conditions:
- The SAS UUT provided certificate has the following valid parameters:
  - Issuer matches a WInnForum approved SAS Provider CA
  - Subject is consistent with the SAS UUT
  - Version = 2 (v3)
  - Serial Number is a positive integer no longer than 20 octets
  - Algorithm ID
  - Validity
     - Not Before is encoded as UTCTime if before the year 2050, or GeneralizedTime for dates in 2050 or later
     - Not After is encoded as UTCTime if before the year 2050, or GeneralizedTime for dates in 2050 or later
  - Subject Public Key Info
     - Public Key Algorithm = SHA-256, SHA-384, or SHA-512
     - Subject Public Key ≥ 2048 bits
  - Subject: Common Name (CN)
  - CA: FALSE
  - Path Length Constraint = None
  - ROLE = SAS
  - FRN = SAS UUT administrator’s assigned FRN
  - Signed by a valid CBRS root of trust
- The SAS UUT provided certificate does not contain the following custom SAS OIDs:
  - ZONE
  - FREQUENCY
  - FCCID
  - SERIAL
  - CPIRID
If any of the above conditions are not met, the SAS UUT FAILS this test. Otherwise, it PASSES.
```

Reply:
This comment will be addressed as part of   the SCS20 test case submission in a new PR.